keanu-project/src/main/java/io/improbable/keanu/distributions/discrete/Multinomial.java

```diff
@@ -123,26 +99,95 @@ private static int draw(KeanuRandom random, Double... categoryProbabilities) {
     }
 
     @Override
-    public DoubleTensor logProb(IntegerTensor k) {
-        long[] expectedShape = p.getShape();
-        TensorShapeValidation.checkAllShapesMatch(
-            String.format("Shape mismatch. k: %s, p: %s",
-                Arrays.toString(k.getShape()),
-                Arrays.toString(expectedShape)),
-            k.getShape(), expectedShape
-        );
-        Preconditions.checkArgument(
-            k.sum(0).elementwiseEquals(this.n).allTrue(),
-            String.format("Inputs %s must sum to n = %s", k, this.n)
+    public DoubleTensor logProb(IntegerTensor x) {
+        if (validationEnabled) {
```

Comment:
I don't really get this validation thing, why not just validate on construction and then only validate the things that change?

e.g: only call:

```
validateBroadcastShapes(x.getShape(), n.getShape(), p.getShape());
validateX(x, n, p);
```

Reply:
Good call. The Multinomial object is never reused after either sample or logProb so I didn't see the point but I've made the change so that we don't force that.